blockchain-albatross/src/blockchain.rs

```diff
@@ -469,11 +468,11 @@ impl Blockchain {
         };
 
         // Public keys are checked when staking, so this should never fail.
-        let slot: IndexedSlot = self.get_block_producer_at(block.block_number(), block.view_number(), Some(&read_txn))
-            .ok_or(PushError::InvalidSuccessor)?;
+        let (slot, _) = self.get_slot_at(block.block_number(), block.view_number(), Some(&read_txn))
+            .unwrap();
```

Comment:
We had `unwrap` at this point before and the code panicked. If I remember correctly though, that only happened because the check for the previous block being present was missing.

Reply:
As long as the previous block (is checked) and the previous macro block exist, we're fine here.

I removed the comment before that line as it doesn't make any sense.